### PR TITLE
 StreamPeer::get_data now returns received size. 

### PR DESCRIPTION
**Pull request by Faless**
Must still return `ERR_*` when `received != p_bytes`.

HTTPClient uses blocking TCP read in thread mode.

Potential fix for #33479 .

**This breaks GDNative compat with 3.1**.

`FileAccessNetworkClient` needs more testing.

Probably too late to merge for `3.2` pushing to `4.0`.

### CODE REVIEW

**Comment by akien-mga** [comment]
@Faless Is this still relevant for the current `master` branch? If so, it needs a rebase.

File: core/io/file_access_network.cpp [hunk 1]
```diff
@@ -145,12 +149,16 @@ void FileAccessNetworkClient::_thread_func() {
...
+				int received;
 				int64_t offset = get_64();
 				uint32_t len = get_32();
 
 				Vector<uint8_t> block;
 				block.resize(len);
-				client->get_data(block.ptrw(), len);
+				err = client->get_data(block.ptrw(), len, received);
+
+				if (err != OK || (uint32_t)received != len)
```
Comment (akien-mga) [inline core/io/file_access_network.cpp:160]: Be sure to add enclosing braces on single-statement `if`s when rebasing.

**Comment by aaronfranke** [comment]
@Faless Is this still desired? If so, it needs to be rebased on the latest master branch.

If not, abandoned pull requests will be closed in the future as announced [here](https://godotengine.org/article/headsup-vulkan-merged-master-branch-unstable).

**Comment by Faless** [comment]
Yeah, I should pick this up again, we need this to fix some cases of HTTPClient with read-until-EOF.

**Comment by Faless** [comment]
Quick update: I really want to do this for `4.0`, hopefully I'll have some time next month. Added to my todo list.

**Comment by Faless** [comment]
Postponed to a later refactor potentially including `File`.

**Pull request closed without merging**